### DISCUSSION TITLE
Irrigation system with grouped switchers

### DISCUSSION THREAD

**Discussion by paulzhdanov** [RavenSystem/esp-homekit-devices#1659] [MEPLHAA Scripts Collection]
Irrigation system with grouped switchers
9.4.2, NodeMCU
```
{
  "c":{"b":[{"g":0,"t":5}],"o":0,"n":"garden","ic":[[14,12,100]],"mc":[[0,32,255]]},
  "a":[{
      "t": 20,
      "s": 0,
      "w": 1,
      "d": 3600,
      "i": 7200,
      "1": {"r":[{"g":100}],"m":[[2,0],[3,0],[4,0],[5,0]]},
      "0": {"r":[{"g":100,"v":1}],"m":[[2,1]]},
      "es":[{
        "t": 20,
        "s": 0,
        "w": 1,
        "d": 3600,
        "i": 7200,
        "1": {"r":[{"g":101}],"m":[[1,0],[3,0],[4,0],[5,0]]},
        "0": {"r":[{"g":101,"v":1}],"m":[[3,1]]} 
      },{
.......................
      }]
  }]
}
```
Here's shrinked version of JSON to show how to create groups of accessories under one button like this:
![IMG_2933](https://user-images.githubusercontent.com/13831213/144508006-fcf52d10-aa8b-4194-b923-5f3a716b357e.jpeg)

I used "m" parameter to send command (it works the same as for "flat" set) to create inching mode and switch on next sprinkler right after previous will run out, it gives you a possibility to create only 1 timer to start watering. That's very useful if you have more than 1 or 2 sprinkler and have to change watering time depends on conditions. Well, native Home app doesn't allow to run watering  on schedule, but 3rd party apps like HomeKit Controller can easily do that. Also I created a group of regular switches for manual control, same way with "es". Unfortunately I didn't find yet how to create soil moisture sensor with RavenSystem, so please share if you'll find a way.

**Comment by runfik** [comment]
Подскажите как вы сделали 4 крана ? скиньте json что делает в вашем примере розетка ? на втором кране ?

**Reply by paulzhdanov** [reply]
Дык в самом первом посте пример json'a, в блоке "es":[{ }] просто сделайте 3 крана (в примере один) и будет вам 4.

**Reply by VaBanck** [reply]
https://discord.com/channels/594630635696553994/913814068803031080/926802085020835890 добро пожаловать

**Reply by runfik** [reply]
> 
Скажите а как работает ваш пример зачем переключатель нужен реверсивный ? для чего вообще хотите использовать ?
